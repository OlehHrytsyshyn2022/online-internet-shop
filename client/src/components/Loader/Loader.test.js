import React from "react";
import { render, screen } from "@testing-library/react";
import CircularLoader from "./Loader";

describe("CircularLoader", () => {
  test("should render a circular progress", async () => {
    render(<CircularLoader />);
    const progress = screen.getByRole("progressbar");
    expect(progress).toBeInTheDocument();
  });

  test("should have a specific size", async () => {
    render(<CircularLoader />);
    const progress = screen.getByRole("progressbar");
    expect(progress).toHaveStyle({
      width: "60px",
      height: "60px",
    });
  });

  test("should have a specific color", async () => {
    render(<CircularLoader />);
    const progress = screen.getByRole("progressbar");
    expect(progress).toHaveStyle({
      color: "#707070",
    });
  });


  test("Snapshot Test CircularLoader", () => {
    const { container } = render(<CircularLoader />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
