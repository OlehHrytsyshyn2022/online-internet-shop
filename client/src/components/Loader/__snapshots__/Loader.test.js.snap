// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularLoader Snapshot Test CircularLoader 1`] = `
<div
  class="MuiBox-root css-95pr9o"
>
  <span
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-ttjza1-MuiCircularProgress-root"
    role="progressbar"
    style="width: 60px; height: 60px;"
  >
    <svg
      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx="44"
        cy="44"
        fill="none"
        r="21"
        stroke-width="2"
      />
    </svg>
  </span>
</div>
`;
